### PR TITLE
Alhimick prog hometask 2

### PR DESCRIPTION
**Pull request by alhimick-prog**
Hometask-2 that requires create program, which finds the middle character(s) in string.

List of changes:

Added:
+ string constant for all messages and for string for exit the program

Deleted:
- extra method `.to_s`
- extra method `validation`
- extra variables

Rework:
* loop implementation from `unless` to `loop`
* control flow from `case` to `if/else`

### CODE REVIEW

File: alhimick-prog/hometask-2/get_middle_charachter.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  if string.length.odd?
+    string[mid]
+  else
+    string[mid - 1] + string[mid]
+  end
+end
+
+def validation(string)
+  if string.empty?
+    puts 'String can not be blank!'
```
Comment (lincAlinc) [inline alhimick-prog/hometask-2/get_middle_charachter.rb]: @alhimick-prog what do you think about adding constant for this string?

File: alhimick-prog/hometask-2/get_middle_charachter.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  else
+    true
+  end
+end
+
+def run_cli
+  string = ''
+  until string == 'exit!'
+    stop = false
+    puts 'Please, enter string:'
```
Comment (lincAlinc) [inline alhimick-prog/hometask-2/get_middle_charachter.rb]: @alhimick-prog what do you think about adding constant for this string?

File: alhimick-prog/hometask-2/get_middle_charachter.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    string = gets.chomp.to_s
```
Comment (lincAlinc) [inline alhimick-prog/hometask-2/get_middle_charachter.rb]: @alhimick-prog why do you add `to_s` method there?

File: alhimick-prog/hometask-2/get_middle_charachter.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+def get_middle(string)
+  mid = string.length / 2
+  if string.length.odd?
+    string[mid]
+  else
+    string[mid - 1] + string[mid]
+  end
+end
+
+def validation(string)
```
Comment (lincAlinc) [inline alhimick-prog/hometask-2/get_middle_charachter.rb]: @alhimick-prog your method return `true` or `false`. In this case method name needs to end with `?`. For example `valid?`

File: alhimick-prog/hometask-2/get_middle_charachter.rb [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  end
+end
+
+def run_cli
+  string = ''
+  until string == 'exit!'
+    stop = false
+    puts 'Please, enter string:'
+    string = gets.chomp.to_s
+    stop = true if string == 'exit!'
```
Comment (lincAlinc) [inline alhimick-prog/hometask-2/get_middle_charachter.rb]: @alhimick-prog `string == 'exit!'` return `true` or `false`. Do we really need `true if` and `stop = false`?

File: alhimick-prog/hometask-2/get_middle_charachter.rb [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    puts "Middle character(s): #{get_middle(string)}" if validation(string) && !stop
```
Comment (lincAlinc) [inline alhimick-prog/hometask-2/get_middle_charachter.rb]: @alhimick-prog Can we add constant for `"Middle character(s):"`?

**Review by lincAlinc — CHANGES_REQUESTED** [review]

**Review by lincAlinc — CHANGES_REQUESTED** [review]
Could you try to rewrite `run_cli` with `loop`?

File: alhimick-prog/hometask-2/get_middle_charachter.rb (hunk 6 above)
Comment (alhimick-prog) [reply alhimick-prog/hometask-2/get_middle_charachter.rb]: Yes, we can. I added constants for all messages.

File: alhimick-prog/hometask-2/get_middle_charachter.rb (hunk 2 above)
Comment (alhimick-prog) [reply alhimick-prog/hometask-2/get_middle_charachter.rb]: Think good. I added constants for all messages.

File: alhimick-prog/hometask-2/get_middle_charachter.rb (hunk 1 above)
Comment (alhimick-prog) [reply alhimick-prog/hometask-2/get_middle_charachter.rb]: Think good. I added constants for all messages.

File: alhimick-prog/hometask-2/get_middle_charachter.rb (hunk 3 above)
Comment (alhimick-prog) [reply alhimick-prog/hometask-2/get_middle_charachter.rb]: I thought it's needed for correct work with user input, but program works correct without this.

Comment (lincAlinc) [reply alhimick-prog/hometask-2/get_middle_charachter.rb]: @alhimick-prog `gets` get back a string. when you use chomp for string - it returns a string.

Comment (alhimick-prog) [reply alhimick-prog/hometask-2/get_middle_charachter.rb]: Yes, this is a lack of knowledge about the method `gets`

**Review by lincAlinc — APPROVED** [review]
Cool!

**Pull request merged**